docs/design-approaches.md

```diff
@@ -0,0 +1,235 @@
+# Design approaches in firecracker-containerd
+
+This document describes possible design approaches for multi-container support with Firecracker and Containerd.
+
+
+## Goals
+
+Support running multiple containers inside a microVM in order to handle Orchestrator tasks with multiple container definitions.
+
+
+## Requirements
+
+1. Orchestrator needs to launch multiple containers in the same VM. The container launch can be staggered. For example, requests could be something like: Launch container A in a VM. After a period of time (once the orchestrator verifies that container A is healthy), launch container B in the same VM.
```

Comment:
nit: I tend to break lines in Markdown around 80 characters as it makes it easier to read in plain-text (doesn't cross the whole screen).  But we're not enforcing that anywhere so it's up to you.